Sprint 15

Fennec-meow opened the pull request.

Fennec-meow committed:
Creating a "CreatingCollectionViewCell":

1. Contains the contents of the collection cell for emoji and color cards

Fennec-meow committed:
Merge branch 'sprint_15' of https://github.com/Fennec-meow/Tracker into habitCell

Fennec-meow committed:
Merge pull request #3 from Fennec-meow/habitCell

Habit cell

Fennec-meow committed:
Creating a "CreatingSupplementaryView":

1. Contains a title for a cell in the collection for emoji and color cards

Fennec-meow committed:
Merge pull request #4 from Fennec-meow/SupplementaryView

Creating a "CreatingSupplementaryView":

Fennec-meow committed:
Changes:

1. Added arrays of "emojis" and "colors";
2. Implemented "collectionView";
3. The "collectionView" methods are configured;
4. A function has been added to CreatingCollectionViewCell to customize the "Emoji" and "Colors";
5. In "CreatingSupplementaryView", the "showNewTracker(title:)" function has been moved to extension.

Fennec-meow committed:
Merge pull request #5 from Fennec-meow/CollectionView

Changes:

Fennec-meow committed:
Changes:

1. The CoreData model has been created;
2. CoreData has been implemented into the project.

Fennec-meow committed:
Merge pull request #6 from Fennec-meow/CoreData

Changes:

IgorChumikov commented on Tracker/AppDelegate.swift:
```diff
@@ -5,32 +5,43 @@
 import UIKit
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
```
final

IgorChumikov commented on Tracker/AppDelegate.swift:
```diff
@@ -5,32 +5,43 @@
-
+    
     func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
         // Called when the user discards a scene session.
```
Можно, удали системные комментарии из AppDelegate.swift. Они не несут полезной информации и только загромождают код.

IgorChumikov commented on Tracker/CoreData/TrackerStore.swift:
```diff
@@ -0,0 +1,202 @@
+        )
+        controller.delegate = self
+        
+        try? controller.performFetch()
```
Инициализацию fetchedResultsController обернуть в do-catch
```
do {
    try controller.performFetch()
} catch {
    assertionFailure("Failed to fetch trackers: \(error)")
}
```
- Ошибка может остаться незамеченной.
- Лучше явно зафейлиться при разработке, чем пропустить баг.

IgorChumikov commented on Tracker/Trackers/CreatingTracker/CreatingCollectionViewCell.swift:
```diff
@@ -0,0 +1,90 @@
+    
+    func configure(with emoji: String? = nil, backgroundColor: UIColor? = nil) {
+        ui.creatingNameLabel.text = emoji
+        if backgroundColor == nil {
```
можно упростить через nil coalescing
```
ui.creatingNameLabel.backgroundColor = backgroundColor ?? .clear

IgorChumikov commented on Tracker/Trackers/CreatingTracker/CreatingCollectionViewCell.swift:
```diff
@@ -0,0 +1,90 @@
+    
+    // MARK: UI components
+    
+    struct UI {
```
Имя структуры UI → ViewComponents или CellUI
 
 UI — слишком абстрактное и общее имя, особенно в больших проектах. Лучше дать более конкретное название.

IgorChumikov reviewed (commented):
Отличная работа! 
 
 Ты большая молодец — вся логика реализована корректно, код хорошо структурирован, читается легко и выглядит профессионально. Приятно видеть, как аккуратно оформлены делегаты, как грамотно организован UI-слой и как отделена работа с данными через Core Data. Использование NSFetchedResultsController, протоколов и делегатов — на высоком уровне 👏

📌 Осталось чуть-чуть до идеала: подправить визуальную часть, чтобы полностью соответствовать дизайну из Figma (например, убрать лишние разделители в таблице и выровнять цвета/отступы). После этого можно смело считать, что работа выполнена на 100% профессиональном уровне.

Продолжай в том же духе! 🚀

Fennec-meow committed:
Changes:

1. Fixed TextFields;
2. Extension+UITextField added;
3. Added a new color to display the SearchTextField on the home screen;
4. Unnecessary divisions in the table between the cells on the "New Habit" and "New Irregular Event" screens have been removed.

Fennec-meow committed:
Merge pull request #8 from Fennec-meow/sprint_15_Fix

Changes:
